### ISSUE TITLE
Enhancements: Example for loading own datasets

### ISSUE THREAD

**Issue by silakanveli** [flyyufelix/cnn_finetune#4]
Very nice work! I found it really easy to test all these networks and build some understanding.

Would be very nice to have example 
1. Howto load own datasets(function)
2. Howto save fine-tuned model

**Comment by flyyufelix** [comment]
Glad to hear that you find the repo useful! To answer your questions:

1. It really depends on your application. Say if you want to load your own dataset and fine-tune with VGG16. You have to implement the load_data() function in vgg16.py that transform your dataset into Keras compatible format (i.e. X_train, Y_train, X_valid, Y_valid):

X_train, Y_train, X_valid, Y_valid = load_data(img_rows, img_cols)

where [X_train, Y_train] represents your training data and the corresponding labels, [X_valid, Y_valid] is your validation set for evaluating performance. 

2. Saving Keras fine-tuned model is very straightforward: https://keras.io/getting-started/faq/#how-can-i-save-a-keras-model

**Comment by arunpatala** [comment]
Do we have to do any preprocessing like normalize the data and subtract mean ?

**Comment by flyyufelix** [comment]
Yes, for RGB dataset, you should subtract the mean pixel of the ImageNet dataset which is mean_pixel = [103.939, 116.779, 123.68] respectively for each channel.

**Comment by jandremarais** [comment]
Are the mean pixel values you gave there not for the channel ordering 'BGR'? In the Keras `preprocess_input` [function](https://github.com/fchollet/keras/blob/master/keras/applications/imagenet_utils.py) for ResNet and VGG, the array first gets reordered from RGB to BGR and then those mean pixel values are subtracted in the order you gave. Or am I missing something? In what order should the color channels be for your pretrained models?

**Comment by flyyufelix** [comment]
Thanks for pointing this out. The channel ordering is 'BGR' indeed. If you use openCV to read the image, channel ordering will be converted to 'BGR' automatically, otherwise you will have to perform the conversion yourself.

**Comment by jandremarais** [comment]
I was also unsure about this. Thanks for clearing it up. And for the awesome repo!

**Comment by EDTejadaG** [comment]
Awsome work!, really easy to understand. I only have some issues with the dimmensions of X_train and Y_train. I assume that I should be careful with the order of the channels depending if I am ussing Theano or Tensorflow right?

**Comment by EDTejadaG** [comment]
Awsome work!, really easy to understand. I only have some issues with the dimmensions of X_train and Y_train. I assume that I should be careful with the order of the channels depending if I am ussing Theano or Tensorflow right?
